WebRoot/public/admin.jsp

```diff
@@ -68,21 +68,26 @@
 	        }
 	    }    
 	}
-	
-    Boolean isLogoff = getParameter(request, "logoff", "0").equals("1");
-    if (isLogoff) {
-        Cookie killAuthToken = new Cookie("ZM_ADMIN_AUTH_TOKEN", null);
-		killAuthToken.setMaxAge(0);
-		killAuthToken.setPath("/");
-		response.addCookie(killAuthToken);
-		String queryString = request.getQueryString();
-        queryString = queryString.replaceAll("&?logoff=1", "");
-		if (queryString.length() > 0)
-			response.sendRedirect(adminUrl + "?" + queryString);
-		else
-			response.sendRedirect(adminUrl); 
-    }	
-	
+%>
+
+<c:if test="${param.logoff eq '1'}">
+	<zm:logout isAdmin="true" />
+
+	<c:set var="req" value="${pageContext.request}" />
+	<c:set var="redirectURL" value="${req.scheme}://${req.serverName}:${req.serverPort}${req.contextPath}" />
```

Comment:
Good point
I have changed code to not recreate url instead used / as redirect url this will redirect to same url hopefully, in case of proxy as well

not able to use adminUrl as it contains '/zimbraAdmin' which is getting appended twice in url